### PR TITLE
Regrouping and formatting of the selection rule specs

### PR DESCRIPTION
**Pull request by dennisdoomen**
**NOTE** Review while hiding whitespace changes.

### CODE REVIEW

File: Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs [hunk 1]
```diff
@@ -12,2045 +12,2318 @@ namespace FluentAssertions.Equivalency.Specs;
...
-        var subject = new
+        [Fact]
+        public void When_a_property_is_an_indexer_it_should_be_ignored()
         {
-            Age = 36,
-            Birthdate = new DateTime(1973, 9, 20),
-            Name = "John"
-        };
+            // Arrange
+            ClassWithIndexer expected = new()
```
Comment (jnyrup) [inline Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs:21]: Is it intentional that you switch from `var foo = new SomeType()` to `SomeType foo = new()`?
As far as I can tell we currently only use taget-type new expressions for local variables five places.

_If_ we want to switch completely to the newly supported syntax, I think we should:
* do it in a separate PR, 
* across the whole repo and
* enforce it via `.editorconfig`.


Using different syntax makes it difficult for contributors to stay consistent.

Comment (dennisdoomen) [reply Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs]: Undid the changes. Will look at that in a follow-up PR.

File: Tests/.editorconfig [hunk 2]
```diff
@@ -1,149 +1,149 @@
 [*.cs]
 
 # IDE0051: Private member is unused
-dotnet_diagnostic.IDE0051.severity = none
+dotnet_diagnostic.ide0051.severity = none
```
Comment (jnyrup) [inline Tests/.editorconfig:4]: Why this change?

Comment (dennisdoomen) [reply Tests/.editorconfig:4]: Rider did that. Apparently that's the convention it used... 🤷🏻

Comment (jnyrup) [reply Tests/.editorconfig:4]: Hmm... their [website](https://www.jetbrains.com/help/rider/Using_EditorConfig.html#what_is) uses uppercase like any other `.editorconfig` I've ever seen.

Comment (dennisdoomen) [reply Tests/.editorconfig]: I removed the entire file from the PR.

**Review by jnyrup — APPROVED** [review]
